plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -197,38 +202,19 @@ func findNodesThatFit(
 }
 
 // Checks whether node with a given name and NodeInfo satisfies all predicateFuncs.
-func podFitsOnNode(pod *api.Pod, meta interface{}, info *schedulercache.NodeInfo, predicateFuncs map[string]algorithm.FitPredicate) (bool, string, error) {
+func podFitsOnNode(pod *api.Pod, meta interface{}, info *schedulercache.NodeInfo, predicateFuncs map[string]algorithm.FitPredicate) (bool, []algorithm.PredicateFailureReason, error) {
+	failedPredicates := make([]algorithm.PredicateFailureReason, 0)
```

Comment:
Please remove it - we don't want to allocate this slice unneceessary, since in most cases it won't be needed.
And append(nil, reasons...) works fine.
So what you want is:
```
var failedPredicates []algorithm.PredicateFailureReason
```